### PR TITLE
[ENH] Add download_dataset function to _data_loaders.py

### PR DESCRIPTION
**Pull request by PaulRabich**
Added a function ``download_dunction`` to the ``_data_loaders.py`` file.
This functions only downloads a dataset without loading it to memory.

### CODE REVIEW

File: aeon/datasets/_data_loaders.py [hunk 1]
```diff
@@ -356,6 +356,52 @@ def _load_saved_dataset(
...
+    ----------
+    name : string,
+            name of the dataset to download
+
+    safe_path : string, optional (default: None)
+            Path to the directory where the dataset is downloaded into.
+
+    Returns
+    -------
+    if successful, string containing the path of the safed file
```
Comment (TonyBagnall) [inline aeon/datasets/_data_loaders.py:374]: saved file

File: aeon/datasets/_data_loaders.py [hunk 2]
```diff
@@ -356,6 +356,52 @@ def _load_saved_dataset(
...
+    """
+
+    Download a dataset from the timeseriesclassification.com website.
+
+    Parameters
+    ----------
+    name : string,
+            name of the dataset to download
+
+    safe_path : string, optional (default: None)
```
Comment (TonyBagnall) [inline aeon/datasets/_data_loaders.py:369]: our standard format is 
safe_path : str or None, default = None

**Review by TonyBagnall — APPROVED** [review]
tiny typo and a formatting request but thanks for this. I think the testing for this needs a bit of a rework, so will add a test for next coverage push.

File: aeon/datasets/_data_loaders.py (hunk 2 above)
Comment (MatthewMiddlehurst) [inline aeon/datasets/_data_loaders.py:369]: Change `safe` in the parameter and documentation to `save`

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
If testing will be done in a different PR, I think my only issue is the docs name as well. Fine with it going in after.

**Review by MatthewMiddlehurst — APPROVED** [review]

**Comment by MatthewMiddlehurst** [comment]
Ignore the failure, unrelated and dealt with in a different PR.

**Comment by PaulRabich** [comment]
Anything else to do?

**Comment by TonyBagnall** [comment]
thanks for this

**Comment by MatthewMiddlehurst** [comment]
@all-contributors add @PaulRabich for code

**Pull request merged**